### ISSUE TITLE
should init command handle package installation?

### ISSUE THREAD

**Issue by itaysk** [hashicorp/terraform#15792]
One of `terraform init`'s  major goals is to fetch provider plugins. Yes, this must happen during project initialization, but fetching plugins also happens regularly during normal development workflow. For example, every time I would use a new provider in my configuration, I would have to run `terraform init` and it feels weird running an *initialization* command on a non-fresh, working project.
To take an example from other systems, look at package managers for different programming languages, for example npm for node.js. The goal is practically the same - fetch and register external plugins. But the wording is better - `npm init` initializes the project and will run only once, and `npm install` installs plugins and can run many times. Also, I can run `install` on an initialized project (for example, cloned from git) and it makes sense.

**Comment by apparentlymart** [comment]
Hi @itaysk! Thanks for this question/feedback.

Agreed that the `init` terminology is a little strange given that this command can also work on an already-initialized directory. The goal was to have a single command that runs all actions so that users don't need to worry about exactly what needs to be done in order to get a working directory ready to use.

So to put it another way, `terraform init` means "get this directory ready to use", which can either be to take it from freshly checked out to functional (by installing modules, plugins) or to adjust an already-initialized directory to match updated config.

Using another verb such as `terraform setup` or `terraform prepare` may have been a better idea; the `init` command has existed for a long time and so it's grown some additional functionality over time. Perhaps eventually we'll migrate to a new verb here, but we won't do that in the near future since the workflow has already changed a lot in recent versions and we'd like to keep things more stable for a while.

**Comment by itaysk** [comment]
Sure, I figured this was the case. Just wanted to bring up this feedback, even if it will never get changed.
